src/lib/eliom_form_sigs.shared.mli

```diff
@@ -0,0 +1,1034 @@
+  | `Number
+  | `Hidden
+  | `Month
+  | `Week
+  | `File
+  | `Email
+  | `Image
+  | `Datetime_local
+  | `Datetime
+  | `Date
+  | `Color
+  | `Button
+  ]
+
+type button_type =
+  [ `Button
+  | `Reset
+  | `Submit
+  ]
+
+module type PARAM = sig
+
+  type +'a elt
+  type +'a attrib
+
+  type uri
+
+  val a_input_required : unit -> Html5_types.input_attrib attrib
+
+  val a_select_required : unit -> Html5_types.select_attrib attrib
+
+  val select_content_of_option :
+    Html5_types.selectoption elt -> Html5_types.select_content elt
+
+  val make_pcdata : string -> Html5_types.pcdata elt
+
+  val make_a :
+    ?a:Html5_types.a_attrib attrib list -> ?href:uri -> 'a elt list ->
+    [> 'a Html5_types.a] elt
+
+  val make_get_form :
+    ?a:Html5_types.form_attrib attrib list -> action:uri ->
+    Html5_types.form_content elt list Eliom_lazy.request ->
+    [> Html5_types.form] elt
+
+  val make_post_form : ?a:Html5_types.form_attrib attrib list ->
+    action:uri -> ?id:string -> ?inline:bool ->
+    Html5_types.form_content elt list Eliom_lazy.request ->
+    [> Html5_types.form] elt
+
+  val cons_hidden_fieldset :
+    Html5_types.input elt list ->
+    Html5_types.form_content elt list ->
+    Html5_types.form_content elt list
+
+  val make_input :
+    ?a:Html5_types.input_attrib attrib list ->
+    ?checked:bool ->
+    typ:[< input_type] ->
+    ?name:string ->
+    ?src:uri ->
+    ?value:string ->
+    unit ->
+    [> Html5_types.input] elt
+
+  val make_button :
+    ?a:Html5_types.button_attrib attrib list ->
+    button_type:[< button_type] ->
+    ?name:string ->
+    ?value:string ->
+    Html5_types.button_content elt list ->
+    [> Html5_types.button] elt
+
+  val make_textarea :
+    ?a:Html5_types.textarea_attrib attrib list ->
+    name:string -> ?value:string ->
+    unit ->
+    [> Html5_types.textarea] elt
+
+  val make_select :
+    ?a:Html5_types.select_attrib attrib list ->
+    multiple:bool ->
+    name:string ->
+    Html5_types.select_content elt ->
+    Html5_types.select_content elt list ->
+    [> Html5_types.select] elt
+
+  val make_option :
+    ?a:Html5_types.option_attrib attrib list ->
+    selected:bool ->
+    ?value:string ->
+    Html5_types.pcdata elt ->
+    Html5_types.selectoption elt
+
+  val make_optgroup :
+    ?a:[ Html5_types.common | `Disabled ] attrib list ->
+    label:string ->
+    Html5_types.selectoption elt ->
+    Html5_types.selectoption elt list ->
+    Html5_types.select_content elt
+
+  val uri_of_string : (unit -> string) -> uri
+
+  val make_css_link :
+    ?a:Html5_types.link_attrib attrib list -> uri:uri -> unit ->
+    [> Html5_types.link] elt
+
+  val make_js_script :
+    ?a:Html5_types.script_attrib attrib list -> uri:uri -> unit ->
+    [> Html5_types.script] elt
+
+  val make_for_attrib : string -> [> `For] attrib
+
+end
+
+module type S = sig
+
+  (* When modifying this interface, please ensure that the ocamldoc is
+   coherent with the ocamldoc from Eliom_uri. *)
```

Comment:
Split the signature and use include ? It's normal include, so it should be linkified properly in ocamldoc.